python/dgl/distributed/kvstore.py

```diff
@@ -901,6 +950,37 @@ def init_data(self, name, shape, dtype, part_policy, init_func):
         self._push_handlers[name] = default_push_handler
         self.barrier()
 
+    def delete_data(self, name):
+        """Send message to kvserver to delete tensor and clear the meta data
+
+        Parameters
+        ----------
+        name : str
+            data name
+        """
+        assert len(name) > 0, 'name cannot be empty.'
+        assert name in self._data_name_list, 'data name: %s not exists.' % name
+        self.barrier()
+        part_policy = self._part_policy[name]
+        num_partitions = part_policy.partition_book.num_partitions()
+        num_clients_per_part = rpc.get_num_client() / num_partitions
+        if self._client_id % num_clients_per_part == 0:
+            request = DeleteDataRequest(name)
+            for n in range(self._group_count):
+                server_id = part_policy.part_id * self._group_count + n
+                rpc.send_request(server_id, request)
+            for _ in range(self._group_count):
+                response = rpc.recv_response()
+                assert response.msg == DELETE_MSG
```

Comment:
how do you delete the array in the backup servers?

Reply:
The array in backup servers are deleted by server.

Reply:
i don't quite understand. the message is sent to every server or just the master server on each machine?

Reply:
It will be sent to every server.